internal/json/json.go

```diff
@@ -0,0 +1,43 @@
+// Copyright Envoy AI Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package json
```

Comment:
centralize the json functions here so that we can switch the impl later (if we ever want to) easily